src/DateFormatService.php

```diff
@@ -4,48 +4,53 @@
 namespace DR\Internationalization;
 
 use DateTimeInterface;
-use IntlDateFormatter;
+use DR\Internationalization\Date\DateFormatHelper;
+use DR\Internationalization\Date\DateFormatOptions;
+use DR\Internationalization\Date\RelativeDateFallbackService;
+use DR\Internationalization\Date\RelativeDateFormatOptions;
 
 class DateFormatService
 {
-    /** @var array<string, IntlDateFormatter> */
-    private array $formatters = [];
-
-    public function __construct(private string $locale, private string $timezone)
-    {
+    private DateFormatHelper $dateFormatHelper;
+    private RelativeDateFallbackService $fallbackHandler;
+
+    public function __construct(
+        private readonly DateFormatOptions $options,
```

Comment:
Also add `private readonly ?RelativeDateFormatOptions  $relativeOptions = null`, to allow setting a global default option